sabnzbd/filesystem.py

```diff
@@ -944,6 +944,45 @@ def remove_all(path: str, pattern: str = "*", keep_folder: bool = False, recursi
 ##############################################################################
 # Diskfree
 ##############################################################################
+def disk_free_macos_clib_statfs64(directory):
+    # MacOS only!
+    # direct system call to c-lib's statfs(), not python's os.statvfs()
+    # Based on code of pudquick and blackntan
+    # Input: directory.
+    # Output: disksize and available space, in bytes
+
+    import ctypes
+    import ctypes.util
+
+    class statfs64(ctypes.Structure):
+        _fields_ = [
+            ("f_bsize", ctypes.c_uint32),
+            ("f_iosize", ctypes.c_int32),
+            ("f_blocks", ctypes.c_uint64),
+            ("f_bfree", ctypes.c_uint64),
+            ("f_bavail", ctypes.c_uint64),
+            ("f_files", ctypes.c_uint64),
+            ("f_ffree", ctypes.c_uint64),
+            ("f_fsid", ctypes.c_uint64),
+            ("f_owner", ctypes.c_uint32),
+            ("f_type", ctypes.c_uint32),
+            ("f_flags", ctypes.c_uint32),
+            ("f_fssubtype", ctypes.c_uint32),
+            ("f_fstypename", ctypes.c_char * 16),
+            ("f_mntonname", ctypes.c_char * 1024),
+            ("f_mntfromname", ctypes.c_char * 1024),
+            ("f_reserved", ctypes.c_uint32 * 8),
```

Comment:
Do we need to declare all the fields, even though we only use a few? From the forum-thread it seems you can also only delcare the field you want to use?

Reply:
AFAIK: No. You really need all the fields ... otherwise the c code is writing to 'unspecified' memory ... hence the coredump I got on Linux with too little parameters.

Reply:
Update: I verified, and yes, with too little parameters, on MacOS I get " Segmentation fault: 11" and on Linux "Segmentation fault (core dumped)". So: all parameters are needed.